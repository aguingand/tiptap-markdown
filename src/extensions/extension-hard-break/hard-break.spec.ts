import { expect, test } from "vitest";
import { parse } from "../../../__tests__/utils";


test('parse markdown', () => {
    expect(parse('example1\\\nexample2')).toMatchSnapshot();
})
test('parse markdown with double spaces', () => {
    expect(parse('example1  \nexample2')).toMatchSnapshot();
})
test('parse markdown with breaks option', () => {
    expect(parse('example1\nexample2', { markdown: { breaks: true } })).toMatchSnapshot();
})
// test('markdown with breaks option + inline', () => {
//     expect(parse('example1\nexample2', { markdown: { breaks: true }, inline: true })).toMatchSnapshot();
// })
test('parse html', () => {
    expect(parse('example1<br>example2')).toMatchSnapshot();
});
