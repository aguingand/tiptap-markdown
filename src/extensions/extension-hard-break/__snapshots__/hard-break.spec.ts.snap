// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "content": [
      {
        "text": "example1",
        "type": "text",
      },
      {
        "type": "hardBreak",
      },
      {
        "text": "example2",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown 1`] = `
[
  {
    "content": [
      {
        "text": "example1",
        "type": "text",
      },
      {
        "type": "hardBreak",
      },
      {
        "text": "example2",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with breaks option 1`] = `
[
  {
    "content": [
      {
        "text": "example1",
        "type": "text",
      },
      {
        "type": "hardBreak",
      },
      {
        "text": "example2",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with double spaces 1`] = `
[
  {
    "content": [
      {
        "text": "example1",
        "type": "text",
      },
      {
        "type": "hardBreak",
      },
      {
        "text": "example2",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;
